bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/FindReplaceLogic.java

```diff
@@ -114,24 +114,45 @@ private void resetStatus() {
 		status = null;
 	}
 
+	@Override
+	public boolean isWholeWordSearchAvailable(String findString) {
+		if (!isWord(findString)) { // $NON-NLS-1$
+			return false;
+		}
+
+		return !isRegExSearchAvailableAndActive();
+	}
+
+	boolean isWholeWordSearchAvailableAndActive(String findString) {
+		if (isWholeWordSearchAvailable(findString)) {
+			return isActive(SearchOptions.WHOLE_WORD);
+		}
+		return false;
+	}
```

Comment:
This method at least needs to be private and the body can be simplified to:
```java
return isWholeWordSearchAvailable(findString) && isActive(SearchOptions.WHOLE_WORD);
```
It may also be sufficient to just inline this condition at its consumers.

Reply:
I have decided that it is reasonable to bundle this logic into a function so as to not introduce added complexity into the functions down below.

Addressed in https://github.com/eclipse-platform/eclipse.platform.ui/commit/a0b224321663e63aa99f60da4a2527f993d68f7d